modules/data/src/dataservices/default-data.service.ts

```diff
@@ -68,67 +70,112 @@ export class DefaultDataService<T> implements EntityCollectionDataService<T> {
       // forward the id of deleted entity as the result of the HTTP DELETE
       map((result) => key as number | string)
     );
   }
 
-  getAll(): Observable<T[]> {
-    return this.execute('GET', this.entitiesUrl);
+  getAll(options?: HttpOptions): Observable<T[]> {
+    return this.execute('GET', this.entitiesUrl, options);
   }
 
-  getById(key: number | string): Observable<T> {
+  getById(key: number | string, options?: HttpOptions): Observable<T> {
     let err: Error | undefined;
     if (key == null) {
       err = new Error(`No "${this.entityName}" key to get`);
     }
-    return this.execute('GET', this.entityUrl + key, err);
+    return this.execute('GET', this.entityUrl + key, err, options);
   }
 
-  getWithQuery(queryParams: QueryParams | string): Observable<T[]> {
+  getWithQuery(
+    queryParams: QueryParams | string | undefined,
+    options?: HttpOptions
+  ): Observable<T[]> {
     const qParams =
       typeof queryParams === 'string'
         ? { fromString: queryParams }
         : { fromObject: queryParams };
     const params = new HttpParams(qParams);
-    return this.execute('GET', this.entitiesUrl, undefined, { params });
+
+    return this.execute(
+      'GET',
+      this.entitiesUrl,
+      undefined,
+      { params },
+      options
+    );
   }
 
-  update(update: Update<T>): Observable<T> {
+  update(update: Update<T>, options?: HttpOptions): Observable<T> {
     const id = update && update.id;
     const updateOrError =
       id == null
         ? new Error(`No "${this.entityName}" update data or id`)
         : update.changes;
-    return this.execute('PUT', this.entityUrl + id, updateOrError);
+    return this.execute('PUT', this.entityUrl + id, updateOrError, options);
   }
 
   // Important! Only call if the backend service supports upserts as a POST to the target URL
-  upsert(entity: T): Observable<T> {
+  upsert(entity: T, options?: HttpOptions): Observable<T> {
     const entityOrError =
       entity || new Error(`No "${this.entityName}" entity to upsert`);
-    return this.execute('POST', this.entityUrl, entityOrError);
+    return this.execute('POST', this.entityUrl, entityOrError, options);
   }
 
   protected execute(
     method: HttpMethods,
     url: string,
     data?: any, // data, error, or undefined/null
-    options?: any
+    options?: any, // options or undefined/null
+    httpOptions?: HttpOptions // these override any options passed via options
   ): Observable<any> {
-    const req: RequestData = { method, url, data, options };
+    let ngHttpClientOptions: any = undefined;
+    if (httpOptions) {
+      ngHttpClientOptions = {
+        headers: httpOptions?.httpHeaders
+          ? new HttpHeaders(httpOptions?.httpHeaders)
+          : undefined,
+        params: httpOptions?.httpParams
+          ? new HttpParams(httpOptions?.httpParams)
+          : undefined,
+      };
+    }
+
+    // If any options have been specified, pass them to http client. Note
```

Comment:
An idea for until we make the breaking change: I think it would be useful we log (`console.warn`) a message when an option is overridden.

Reply:
Agree, I'll add this now

Reply:
@timdeschryver do these look ok?
1. All usage of getWithQuery:
`Warning: The queryParams parameter of getWithQuery will be removed in next major release.`

2. Where queryOptions and HttpOptions are both used:
Warning: HttpOptions.httpParams will be merged with `queryParams` when both are are provided. In the event of a conflict HttpOptions.httpParams will override queryParams`.

Reply:
2 get's my vote 😀
It can also include `The queryParams parameter of getWithQuery will be removed in next major release`

Reply:
Okeydoke, adding now, thanks Tim

Reply:
done